### DISCUSSION TITLE
Can the energy-only model in nequip v0.11.0 predict forces?

### DISCUSSION THREAD

**Discussion by Mehryar-jn** [mir-group/nequip#510] [General]
I noticed that the energy-only model in nequip v0.11.0 does not predict forces for the test data in the file generated by the callback TestTimeXYZFileWriter. However, custom dataset statistics, loss, and validation metrics for energy-only training will predict forces for the test set. 
I'm wondering if there is a way to predict forces using the energy-only model in nequip v0.11.0. Do we get forces by nequip-compile on an energy-only model?

**Comment by Linux-cpp-lisp** [comment]
As in you want to train only on energies but predict energies and forces? I would not necessarily expect very good forces in this case

**Reply by Mehryar-jn** [reply]
Yes, train only on energies and predict both energies and forces. I agree that it's not very accurate, but I expect the predicted forces to be a reasonable approximation of the actual forces. Doesn't nequip calculate the gradient of total energy with respect to particle positions and predict that as forces in training on energy-only situations?

**Reply by cw-tan** [reply]
You just have to use the regular model that predicts forces too then: https://nequip.readthedocs.io/en/latest/api/nequip_model.html#nequip.model.NequIPGNNModel

It just means you'll have to accept the extra flops and walltime computing forces at every training step since it's not used in a loss, or for any metrics except at test time. The other thing one could do is train an energy-only model, then load the energy-only model weights to a freshly built force model, and use that model for a `test` only run, but that will be slightly annoying.

**Reply by Mehryar-jn** [reply]
I see. Great! Thanks @cw-tan
